streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedSerializer.java

```diff
@@ -56,27 +56,28 @@ private static boolean isUpgrade(final Map<String, ?> configs) {
             return false;
         }
 
-        switch ((String) upgradeFrom) {
-            case StreamsConfig.UPGRADE_FROM_0100:
-            case StreamsConfig.UPGRADE_FROM_0101:
-            case StreamsConfig.UPGRADE_FROM_0102:
-            case StreamsConfig.UPGRADE_FROM_0110:
-            case StreamsConfig.UPGRADE_FROM_10:
-            case StreamsConfig.UPGRADE_FROM_11:
-            case StreamsConfig.UPGRADE_FROM_20:
-            case StreamsConfig.UPGRADE_FROM_21:
-            case StreamsConfig.UPGRADE_FROM_22:
-            case StreamsConfig.UPGRADE_FROM_23:
-            case StreamsConfig.UPGRADE_FROM_24:
-            case StreamsConfig.UPGRADE_FROM_25:
-            case StreamsConfig.UPGRADE_FROM_26:
-            case StreamsConfig.UPGRADE_FROM_27:
-            case StreamsConfig.UPGRADE_FROM_28:
-            case StreamsConfig.UPGRADE_FROM_30:
-            case StreamsConfig.UPGRADE_FROM_31:
-            case StreamsConfig.UPGRADE_FROM_32:
-            case StreamsConfig.UPGRADE_FROM_33:
-            case StreamsConfig.UPGRADE_FROM_34:
+        switch (StreamsConfig.UpgradeFromValues.valueOf("UPGRADE_FROM_" + ((String) upgradeFrom).replace(".", ""))) {
```

Comment:
Could you move this to the enum by adding the following static method?
```java
public static UpgradeFromValues getValueFromString(final String upgradeFrom) {
    return UpgradeFromValues.valueOf("UPGRADE_FROM_" + upgradeFrom.replace(".", ""));
}
```

Reply:
Sure.

Btw: I also move the new enum into it's own file, and move it to package `internals` -- there is no reason to expose it via `StreamsConfig`, it also works, and we don't have to worry about changing public API without a KIP.